### ISSUE TITLE
Interface slow and laggy, latency between cursor and brush

### ISSUE THREAD

**Issue by tushantin** [opentoonz/opentoonz#1371]
I have a fairly good system, with AMD FX 8320 Eight-Core Processor, GTX 750 Ti card and 16 gigs of RAM. I'm running Ubuntu 16.04 LTS. 

As such, usually it takes quite a lot of vertices and vector-nodes to lag my system, but for some reason this happens to me in OpenToonz 1.1.3 on a blank canvas (even if it's fairly light on system resources). It doesn't lag my system or anything, just that drawing on it (or even resizing the interface for that matter) is slow as hell. 

I think this video I recorded should describe my issue better than I can in words: https://youtu.be/yKh6RE2_ISI

I've tried switching my driver to Nouveau (opensource) and back to proprietary, but the problem persists in both cases. So I'm not entirely sure if the bug is in the drivers or the software itself (as I mentioned before, no other software, not even Inkscape, gives me such an issue). I also have no idea how to even work around this issue. So I'd be very grateful for any kind of help. 

I know there is a separate post about a similar issue somewhere here, but while the issue is similar it didn't seem to be identical to mine — that is, there was no mention of laggy interface — which is why I thought I'd make a separate post altogether. 

From the video, you may also notice another bug regarding levels, but I think I'll make a separate post about it.

**Comment by morevnaproject** [comment]
I see that you are running OpenToonz 1.1.3 Appimage (Morevna Edition).

I can reproduce the same issue at one of my workstations (NVidia GTX 650, Proprietary driver, Ubuntu 16.04).

There could be two possible reasons for this problem:
1. The problem is related with AppImage package
2. or, the problem is related with driver.

I will do some tests tomorrow to dig the problem deeper.

**Comment by morevnaproject** [comment]
@turtletooth Can you please add "Linux" label to this issue? ^__^

**Comment by WPFilmmaker** [comment]
Hi I am running OpenToonz 1.1.3 Appimage (Morevna Edition). as well. I am running it on Lubuntu 16.04.3 on a 4x Intel(R) Core(TM) i7-6500U CPU @ 2.50GHz 12GB of ram with the latest opensource drivers.

However I am unable to reproduce the slowness. Is someone else using Lubuntu?

**Comment by morevnaproject** [comment]
@WPFilmmaker Please let me know your videocard model. The problem seems to be affecting only some NVidia cards - I have appimage working correctly on my other laptop with Intel videocard. Also works fine on other laptop with NVidia mobile card.

Additional information regarding my workstation (which is affected y issue): NVidisa driver version is 370.28.

@tushantin what is your (proprietary) driver version?

**Comment by tushantin** [comment]
@morevnaproject I'm running this (version 375.66) 
![image](https://user-images.githubusercontent.com/5314015/28973523-119f2c56-7951-11e7-9941-2b59a9549b11.png)

**Comment by morevnaproject** [comment]
@tushantin Thank you!

@blackwarthog I know you have NVidia card on Debian. Can you please check if you affected by the same issue as @tushantin shows on video? - https://youtu.be/yKh6RE2_ISI
(please check both Appimage and native build). ^__^

**Comment by blackwarthog** [comment]
At my system all works fine with native and appimage build. I cannot reproduce the bug. (debian8, nv760)

**Comment by WPFilmmaker** [comment]
@morevnaproject I have a GeForce 940M and I am offered the same version for the proprietary drivers as @tushantin but I am using the opensource ones. Also unlike him, my processor is Intel and I don't run the proprietary processor microcode.

**Comment by morevnaproject** [comment]
@WPFilmmaker Thank you very much for clarification! I have Intel processor as well, but I am affected by issue.

@tushantin Please disconnect graphic tablet from your workstation, restart OpenToonz and try to resize panels - does issue still happen? ^__^

**Comment by tushantin** [comment]
@morevnaproject Incredible, now it doesn't seem to lag! (Well, it has a very slight latency issue even now, but not something that isn't manageable, but at least the slowness is gone.) But now, even after plugging the tablet back in again and restarting Opentoonz doesn't give me that issue though the pressure-sensitivity seems to have gone (it's working in MyPaint though).

**Comment by morevnaproject** [comment]
@tushantin Great, that's good to know. ^__^ For me the lag is also happens only when tablet is connected. What is the model of your tablet?

> But now, even after plugging the tablet back in again and restarting OpenToonz doesn't give me that issue though the pressure-sensitivity seems to have gone

Please make sure you have  Max size for brush set big enough.
![ot-config6](https://user-images.githubusercontent.com/332868/28996269-0c9fd332-7a27-11e7-9566-aea1dee316c1.png)

**Comment by tushantin** [comment]
@morevnaproject Yup, still an issue. Looks like I might need to restart or something. :P I don't know why this is happening though. Is it the Wacom drivers causing this? Is it the processor microcode or something? 

As for my model, I'm using a (probably discontinued) Wacom Bamboo One 5x8 Pen (or as it says in the Wacom setting, "Wacom Bamboo1 5x8 Pen"). I still use it because it has served me well for quite a long time now. 

![image](https://user-images.githubusercontent.com/5314015/28996349-60c757ec-7a1c-11e7-9e0d-4abcacd7fae5.png)

**Comment by tushantin** [comment]
Oh, one more thing I forgot to mention, which might be an important information (I don't know). For some reason, regardless of whether pressure sensitivity was active, while my mouse interacts with OpenToonz well — that is moving the brush around without painting, or hovering over tool icons for their labels to show — I can't seem to do the same with my tablet sometimes. When I hover over any tool with my tablet, nothing shows as if the software doesn't seem to detect my tablet until I press / tap it.

**Comment by tushantin** [comment]
Sorry for triple post, but just wanted to make an update on something I've noticed today. Now the software can't seem to detect my tablet (hence no pressure-sensitivity), but because the tablet is plugged in I'm still getting the slowness and lag.

**Comment by morevnaproject** [comment]
@tushantin I assume your tablet is plugged into front USB port. Can you please try to plug it into rear USB port, reboot, start OpenToonz and then check how it behaves?

**Comment by tushantin** [comment]
@morevnaproject Okay, it's in the back port, though I seem to have got the pressure-sensitivity working before I switched. Strangely, though, all other problems seem to still persist (including the slowness and tablet not being able to hover-display labels on icons).

**Comment by morevnaproject** [comment]
Oksy, thank you for checking. At the moment I am a bit puzzled. I can reproduce the same issue at one of my workstations. But I have other workstations (laptops) with exactly same OS (cloned via partimage) and connecting same graphic tablets (I have tested several) do not trigger this issue on them.

So, I am still not sure what triggers that issue.

As a note to myself, here is a list of things I have to verify:
- [ ] Try to plug tablet into back USB port and test if issue persists.
- [ ] Try to uninstall Nvidia drivers from affected workstation (it is possible they causing some conflict).
- [ ] Try to do full system update on affected workstation using "apt-get upgrade" and "apt-get dist-upgrade".
- [ ] Test if issue happens with different Linux distro (Debian?) on this hardware.
- [ ] Clone image to other workstation and test the issue there.

I will need some time to check all those possibilities. The machine I am testing is heavily used by other studio members, so I am a bit limited with timeframe for my experiments. Please feel free to ping me within a week if I will not come up with any news. ^__^

**Comment by blackwarthog** [comment]
Also we need to compare USB device info (for example usb-mode v2.0 or v1.2). For that run the following commands:

```
~$ sudo lsusb

Bus 001 Device 002: ID 8087:8000 Intel Corp. 
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 003 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 002 Device 006: ID 056a:0010 Wacom Co., Ltd Graphire
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
...
```

Look to record "Bus **002** Device **006**: ID 056a:0010 **Wacom** Co., Ltd Graphire". And make following device path: "/dev/bus/usb/**002**/**006**" (/dev/bus/usb/\<bus-number\>/\<device-number\>) to run lsusb again:

```
~$ sudo lsusb -D /dev/bus/usb/002/006

Device: ID 056a:0010 Wacom Co., Ltd Graphire
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               1.00
  bDeviceClass            0 (Defined at Interface level)
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0         8
  idVendor           0x056a Wacom Co., Ltd
  idProduct          0x0010 Graphire
  bcdDevice            1.20
  iManufacturer           1 WACOM
  iProduct                2 ET-0405-UV1.2-0
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength           34
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0x80
      (Bus Powered)
    MaxPower               40mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.00
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      98
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval              10
Device Status:     0x0000
  (Bus Powered)

```

**Comment by blackwarthog** [comment]
Also we may try to look and compare usb-bus activity https://www.mjmwired.net/kernel/Documentation/usb/usbmon.txt

**Comment by tushantin** [comment]
Okay, so this is what I got. Not sure how helpful that may be... 

`cyan@cyan-desktop:~$ sudo lsusb -D /dev/bus/usb/004/002
Device: ID 056a:006b Wacom Co., Ltd 
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 (Defined at Interface level)
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x056a Wacom Co., Ltd
  idProduct          0x006b 
  bcdDevice            1.00
  iManufacturer           1 Wacom Co.,Ltd.
  iProduct                2 CTE-660
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength           34
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0x80
      (Bus Powered)
    MaxPower               98mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.00
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     146
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               4
Device Status:     0x0000
  (Bus Powered)
`

**Comment by tushantin** [comment]
Okay, so this is what I got. Not sure how helpful it might be though... 

```~$ sudo lsusb -D /dev/bus/usb/004/002
Device: ID 056a:006b Wacom Co., Ltd 
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 (Defined at Interface level)
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x056a Wacom Co., Ltd
  idProduct          0x006b 
  bcdDevice            1.00
  iManufacturer           1 Wacom Co.,Ltd.
  iProduct                2 CTE-660
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength           34
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0x80
      (Bus Powered)
    MaxPower               98mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.00
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     146
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               4
Device Status:     0x0000
  (Bus Powered)
```

**Comment by tushantin** [comment]
Here's an update: Just out of curiosity, I tried switching to the "other" proprietary nvidia driver I had in the screenshot I posted before (340.102), restarted, and tested OpenToonz. The problem still seems to be there. 

I'll do one thing, I'll update my nvidia drivers (from the download link from nvidia website) and see if that has any effect. Now, if I can only figure out how to install this .run file...

**Comment by morevnaproject** [comment]
Hello @tushantin ! I am strongly discourage from installing NVidia drivers from .run file - this will bring a mess into your system and will be hard to remove. The problem seems to be unrelated with NVidia.

**Comment by tushantin** [comment]
@morevnaproject Sorry I wasn't able to ping you in a week's time, so thought I'd do so now. Any news on your testing?

**Comment by tushantin** [comment]
WHAAAAAAAAAAAAAAAAAAAAAT?!?!?!? 

I have no idea what on earth just happened, but the bug seems to have fixed itself out of the blue! But I think I might have some clue about it. See, just today I decided to [install Budgie desktop](http://www.omgubuntu.co.uk/2017/08/install-budgie-desktop-10-4-on-ubuntu) on top of my Ubuntu system, [as well as restore the Unity greater,](https://askubuntu.com/questions/403477/restore-unity-greeter) just to test the new desktop out because I'm a tinkerer like that. And when I started Opentoonz, lo an behold, the problem is no longer there, even WITH the tablet connected! 

(Interestingly, Budgie's notification seemed to tell me that it can't recognize my "Wacom tablet" and it may not work as intended, but that's probably besides the point.) 

Something tells me that there might have been something interfering with the tablet interfacing with Opentoonz (and that app specifically) when using Unity, XFCE, LXDE and Gnome, but for whatever reason that something doesn't seem to be doing so on Budgie. 

Does that info help in any way to figure out how to diagnose the issue and create a work-around to a problem like this? Is there something I can test out? 

https://youtu.be/N5EUR-3839M

**Comment by morevnaproject** [comment]
Hello @tushantin!

Sorry for delayed reply.

I also did several tests on my hardware. I have discovered the following:
**The bug is only happens when you have tablet connected while system is booting.**

With the following steps the bug doesn't happens:

- disconnect tablet
- reboot PC, login to desktop environment (tested with KDE and LXDE)
- connect tablet
- start OpenToonz

After that I have full pressure sensitivity and no lag.

After system (and desktop environment) is fully booted you can connect and disconnect tablet several times without a harm. But if the tablet connected on boot, then most likely you will have a lag.

The real behavior is dependent from tablet model. I have tested three of them (pretty old):
1. Wacom Bamboo CTE-650 - If your tablet was connected on boot, but you will disconnect it and connect again (while OT is closed), then the bug doesn't happen.
2. Wacom Bamboo CTH-470 - Disconnecting and re-connecting the tablet doesn't helps. I have to re-boot my PC without a tablet to avoid the bug.
3. Wacom Graphire CTE-640 - Disconnecting and re-connecting fixes the bug.

Conclusion: **It seems Ubuntu 16.04 does some tablet configuration on boots, which interferes with OT functionality. Make sure your tablet is unplugged when workstation is booting.**

**Comment by tushantin** [comment]
@morevnaproject I'm running **CTE-660**. Does the issue still occur if you change USB ports?

Given your tests I tried to run the same, but something strange just happened: I no longer experience the issue, regardless of whether I plug the tablet in before or after booting, and I have no idea why. 

The only changes I've made to my system since are the following: 
1) Run system updates (which previously failed to fix the issue), 
2) Install Arc Theme and Papirus Icon theme, 
3) Install Budgie desktop environment
4) Change the port that the tablet was plugged into (previously it was at the back, now it's at the front so I can easily take it out and put it back in). 

I can confirm that this issue neither affects me on Budgie environment nor Unity now, which is weird, because it looks like Budgie is influencing Unity in some way. The Arc theme I installed on Budgie is carried on to Unity, and Unity also now seems to use alternative applications such as Files (which is used by Budgie) rather than its previous defaults like Nautilus. 

I have a strong feeling that there may have been something in the Unity / Gnome desktops that seem to be interfering with the tablet interfacing with OT, but seems to have either gone dormant or entirely replaced by Budgie defaults. And even then, this issue only seemed to occur IN OT. 

Which makes me a bit curious: I'm not a programmer or anything, so I don't know much about how software gets developed, but is there some unique framework that OT uses that isn't used by other general GTK / KDE applications?

**Comment by tushantin** [comment]
UPDATE: Sorry for double-post, but now it seems I'm having a different sort of issue entirely, but apparently for the same reasons. Now when I boot into Unity directly (rather than simply switching desktop environments) and start OT with tablet connected, OT starts up fine then quickly crashes and disappears; it stays up when I unplug the tablet before starting it. 

But that's not the case when booting directly in Budgie (rather than simply switching desktop environments via log-out); OT runs fine regardless of whether the tablet has been connected.

**Comment by tushantin** [comment]
ANOTHER UPDATE: Okay, I'm not entirely sure what exactly changed. Since I mentioned the OT doesn't give me any issues on Budgie, unlike other environments, I exclusively used OT and Budgie to finish an entire project in. Now, just when I thought I could actually do another project, OT now once again gives me the same issues. This time in Budgie too. Same issue as the last time in other environments. 

I have no idea why, and what I can potentially do about this.

**Comment by morevnaproject** [comment]
Hello @tushantin ! Please use the workaround I mentioned above: 

> **Make sure your tablet is unplugged when workstation is booting.**

**Comment by tushantin** [comment]
@morevnaproject For some reason the issue vanished again after I replugged the tablet into the front USB slot (earlier I put it in the back). It looks like you may be right; part of the reason may actually be the difference in USB slots (back vs front). 

Though, I'm still on Budgie. I'll try it again in Unity / Gnome or some other environment later, and also give that workaround you mentioned a shot.

**Comment by Toybulldog** [comment]
Hello tushantin

Maybe I can help
I found out with me, that I can not use the driver : X.Org X server -- Wacom input driver
![x org x server -- wacom input driver](https://user-images.githubusercontent.com/11164330/31893434-d292aa52-b80b-11e7-9f43-43dd34391c68.png)





i use Wacom CTH-680 , OpenToonz-1.1.3-morevna-2017.08.03-97dd6-linux-64bit

My System Budgie Ubuntu 17.10

Amd Fx-8370e eight-core processeor
AMD Radeon rx 460 graphics


But if I use the current githup of (OpenToonz), then that plays no role .

**Comment by tushantin** [comment]
@Toybulldog Hmm, I just checked with my Synaptic, and it looks like the version I'm using is 1:0.32.0-0ubuntu3 (which "seems" to be an older version, compared to yours, which is understandable since I'm using LTS release). 

Are you having the same issue? Are there are differences in your system compared to mine and Morevna?

**Comment by morevnaproject** [comment]
@Toybulldog Thank you very much for the hint! I will test it on my machine.

@tushantin Toybulldog suggest to remove  xserver-xorg-input-wacom package to eliminate the issue. The version number isn't really matter.

**Comment by Toybulldog** [comment]
Hello morevnaproject

Watch the video then you'll see again what I mean

https://vimeo.com/239614492

lg peter

**Comment by jpturcotte** [comment]
@Toybulldog The video is private, so it will not show.

**Comment by Toybulldog** [comment]
Sorry for that .
 Now it should be for everyone to see .

I have still forgotten to say in the video .
After you have installed or uninstall the driver, you have to log out of your desktop .

**Comment by jpturcotte** [comment]
@Toybulldog I think you should report your particular problem directly to the Morevna devs. If I am not mistaken, that would be there: https://github.com/morevnaproject/morevna-club/issues.

**Comment by tushantin** [comment]
@jpturcotte I'm not entirely sure about that, man. From what I see in the video, while certain issues after the installation of the wacom driver seems to exist in Morevna specific build, the same slowdowns of the interface still exist in the Github version too. 

@Toybulldog Very thanks for the video! That was pretty enlightening. 

If I'm not mistaken, could this be a case where Opentoonz may be having difficulty in working with the stable Wacom driver, which causes the slowdowns (and, in my case, also crashes whenever I try to switch between the use of Wacom and Mouse, even if it's not slowing down)? Because that could explain why everything else seems to be working except for this software. 

Or is this something to do with the some fault in the driver itself, which apparently the dev-version of the driver seems to have fixed? 

I'll do one thing, I'll try both. I'll not only install the Github version and see how that fairs in my build, but also whether switching the drivers works well for me. 

@morevnaproject BTW I tried your work-around. Disconnecting the tablet while booting, then reconnecting after booting. Now OpenToonz won't even launch. :/

**Comment by artisteacher** [comment]
@tushantin Does it make a difference whether you reconnect after launching OpenToonz? Or do you have multiple graphics cards? If I have automatic graphics switching enabled on my Mac, I can't open OT #1439 unless I have some other memory intensive programs open at the same time - like Photoshop.

**Comment by tushantin** [comment]
@artisteacher No, no difference, apparently. Right now, I just rebooted with the tablet connected, and it starts up fine now (but still with that same crashing problem as before; that is, when I work with the mouse, switch to tablet — both connected at the same time — then switch to mouse again, it is prone to crashing often). 

Hmm, I don't think I tried OpenToonz before I upgraded my RAM, so I'm not entirely sure about the memory problem, but that sounds like a pretty messy issue of its own...

**Comment by tushantin** [comment]
@morevnaproject @Toybulldog Uh oh. Looks like attempting to remove the driver gave me this warning. Now, I don't mind removing Lubuntu, but I'm a bit scared of proceeding with the removal of whatever xserver input thingy it means. @_@

![selection_815](https://user-images.githubusercontent.com/5314015/31967184-c4076034-b92a-11e7-9258-e845febd80cb.png)

**Comment by morevnaproject** [comment]
> @Toybulldog I think you should report your particular problem directly to the Morevna devs. If I am not mistaken, that would be there: https://github.com/morevnaproject/morevna-club/issues.

@jpturcotte His video is not about reporting a problem. With this video he is kindly providing the solution to avoid the issue, which is reported by thread author.

The https://github.com/morevnaproject/morevna-club/issues is for handling our internal studio issues and intended for Russian-speaking audience only. ^__^

**Comment by morevnaproject** [comment]
@tushantin 

> Uh oh. Looks like attempting to remove the driver gave me this warning.

It is safe to remove those packages. Those packages are just placeholders to handle installing all drivers (they are also called meta-packages). This is clearly indicated in  the description for each package:

![screenshot_003](https://user-images.githubusercontent.com/332868/32212058-d15fc7cc-be47-11e7-8ed5-56670cd741ab.png)

**Comment by morevnaproject** [comment]
About the origins of the problem. I strongly suspect that the problem is related with QT. Possibly tweaking some compilation options for QT can help.

**Comment by tushantin** [comment]
Okay, I tried removing the driver as mentioned. It works! And not only does it work, but it "feels" less clunky and a whole lot more stable now! Plus, OT seems to crash neither when I switch between tablet and mouse nor when I disconnect and reconnect the tablet while OT is still running — and it still has pressure-sensitivity! Huzzah! 

There IS a caveat, however (not "caviar", those things are expensive). When on a hunch I checked Settings > Wacom, it gave me the error in the following attached image. I wonder if @Toybulldog also has the same issue, or if it's just with my tablet? 

Because if this is a common trade-off issue, then removing the non-working driver seems to be a great temporary fix, though we'd still have to figure out what's causing this issue in the first place (where if it's the driver itself or QT compilation options as you said). 


![wacom tablet_833](https://user-images.githubusercontent.com/5314015/32219083-06f51f08-be53-11e7-840f-ada110d498ca.png)

**Comment by jpturcotte** [comment]
Sorry for the misunderstanding, folks! @morevnaproject I hope I have not caused you trouble.

**Comment by mobilkitty** [comment]
hi
I solved my own issues with slow cursor updates by switching graphics tablet from USB 3 port (
accidentally and silently it was there  :)  ) to usb 2 without any reboots
and now it works great
Arch linux with latest upgrades

**Comment by tushantin** [comment]
UPDATE: It seems like the laggyness has returned, despite not using xorg-input-wacom and directly using libwacom (yes, even in USB 3 port). o_O I dunno what happened, so I think it might be the ~~work of an enemy stand~~ recent updates. But I can't say for sure or don't know what it is.

**Comment by tushantin** [comment]
Okay, some update. I'm using the latest OpenToonz now, and my system is in the same state as it was previously (that is, I removed the apparently "troublesome" driver, and using the "working" one as mentioned by Toybulldog). 

While I'm not experiencing any slowdowns, I'm having a completely different issue, but I think it might be related to this one: Using the Middle-Click to pan around the canvas and Left-Click to draw works fine, but using the same Middle-Click on my tablet doesn't seem to work. I don't know what might be causing this, but I think it might be the same source that caused lag earlier (that is, the way OpenToonz seems to work with Linux wacom drivers). 

Another issue I found myself facing after removing that driver last time is that, for some reason, GIMP application doesn't seem to be responsive to me anymore. In other words, I don't think removing that driver might be a solution, even if it is a sort-of functional workaround for OT. 

I'll reinstall the previous driver and see what happens. I'll just update this post after checking.

**Comment by artisteacher** [comment]
I get that disappearing behavior in OS X when the stroke has been interrupted by auto-save or an unintentional button press on the stylus.

**Comment by b4zz4** [comment]
Linux Mint 18 (Sarah)
i   libwacom-bin    
i   libwacom-common  
i   libwacom2 
i   xserver-xorg-input-wacom  


![wacom opentoonz](https://user-images.githubusercontent.com/643833/34121579-514801f8-e408-11e7-9539-a67e04ec292b.gif)

Not work :(

**Comment by tushantin** [comment]
@b4zz4 Yes, that is EXACTLY the issue I'm having!

**Comment by morevnaproject** [comment]
@tushantin Please let me know which version are you using

**Comment by tushantin** [comment]
@morevnaproject OT 1.2.0.1, Morevna Edition.

**Comment by morevnaproject** [comment]
Ah, I see. We changed QT version, this is might be the reason why tablet lag has gone.

Regarding the other tablet bug - this is a completely different problem. Can you try this special build (despite the version number it is almost the same as 1.2.0.1) - https://dev.icystar.com/downloads/OpenToonz-1.1.3.1-morevna-17.11.30-linux64-76dc6.appimage

**Comment by tushantin** [comment]
Alright will do, but Google seems to tell me that the source is unsafe. Any idea why?

**Comment by morevnaproject** [comment]
THat's SSL problem. If you don't trust the source you can get it from here - https://sourceforge.net/projects/opentoonz-morevna-edition/files/stable/1.1.3.1/

**Comment by tushantin** [comment]
Okay, that build seems to work great! I can draw, select and transform fairly well. Feels much more stable, even more so than the previous release. The middle-click on the tablet-stylus still doesn't work though. :/ I dunno why, because it works on my browser.

**Comment by morevnaproject** [comment]
The middle-click is a different problem.

If the tablet lag has gone, then I suggest to close this issue. ^__^

**Comment by tushantin** [comment]
Thank you very much, @morevnaproject ! 

@artisteacher Please check if the OSX build in the link provided by Morevna works for you and solves your issue.

**Comment by artisteacher** [comment]
@tushantin, I couldn’t open that build due to a missing library. The latest build however doesn’t seem to have the problem with the interrupted stroke due to auto-save! Accidental button-press during a stroke continues to have that undesirable behavior. and sometimes the stroke width is incorrect.

**Comment by morevnaproject** [comment]
@tushantin I have made a special test build for you - https://dev.icystar.com/downloads/OpenToonz-1.2.0.1-morevna-17.12.22-linux64-5dd88.appimage
Can you please check if it has tablet bug on your system?

I mean this bug:

>  Using the Middle-Click to pan around the canvas and Left-Click to draw works fine, but using the same Middle-Click on my tablet doesn't seem to work. I don't know what might be causing this, but I think it might be the same source that caused lag earlier (that is, the way OpenToonz seems to work with Linux wacom drivers).

**Comment by tushantin** [comment]
@morevnaproject Okay, tested, and sadly I still can't seem to pan around via middle-click with my tablet stylus. I can do it just fine with the mouse. Would it be fair to assume that, perhaps, the middle-click problem is related to the stylus-cursor not showing tool-tips or appropriate cursor styles when hovering over icons, like we usually get when we use the mouse in that way? 

Oh yeah, now that I think about it, I also can't seem to Right-Click via the stylus on OT! Forgot to mention that since I barely use the Right-Click.

**Comment by morevnaproject** [comment]
@tushantin Thank you for testing! I have opened a new issue for your problem - #1708
I think it will be more logical to continue discussion there.

**Comment by morevnaproject** [comment]
Please close this issue.

**Comment by morevnaproject** [comment]
@tushantin or @turtletooth Please close this issue.

**Comment by tushantin** [comment]
@morevnaproject Shit, I don't know how to do it. Lemme see...
